spec.html

```diff
@@ -48962,88 +48934,89 @@ <h1>agent-order</h1>
       </emu-note>
 
       <emu-note>
-        <p>~init~ events do not participate in synchronizes-with, and are instead constrained directly by happens-before.</p>
+        <p>In a candidate execution _execution_, ~init~ events do not participate in this Relation and are instead constrained directly by happens-before.</p>
       </emu-note>
 
       <emu-note>
-        <p>Not all ~seq-cst~ events related by reads-from are related by synchronizes-with. Only events that also have equal ranges are related by synchronizes-with.</p>
+        <p>In a candidate execution _execution_, not all ~seq-cst~ events related by reads-from are related by synchronizes-with. Only events that also have equal ranges are related by synchronizes-with.</p>
       </emu-note>
 
       <emu-note>
-        <p>For Shared Data Block events _R_ and _W_ such that _W_ synchronizes-with _R_, _R_ may reads-from other writes than _W_.</p>
+        <p>For Shared Data Block events _R_ and _W_ in a candidate execution _execution_ such that _W_ synchronizes-with _R_, _R_ may reads-from other writes than _W_.</p>
       </emu-note>
     </emu-clause>
 
-    <emu-clause id="sec-happens-before" aoid="happens-before">
+    <emu-clause id="sec-happens-before">
       <h1>happens-before</h1>
-      <p>For a candidate execution _execution_, _execution_.[[HappensBefore]] is the least Relation on events that satisfies the following.</p>
+      <p>For a candidate execution _execution_, its <dfn>happens-before</dfn> Relation is the least Relation on events that satisfies the following.</p>
 
       <ul>
-        <li>For each pair (_E_, _D_) in _execution_.[[AgentOrder]], _execution_.[[HappensBefore]] contains (_E_, _D_).</li>
-        <li>For each pair (_E_, _D_) in _execution_.[[SynchronizesWith]], _execution_.[[HappensBefore]] contains (_E_, _D_).</li>
-        <li>For each pair (_E_, _D_) in SharedDataBlockEventSet(_execution_), _execution_.[[HappensBefore]] contains (_E_, _D_) if _E_.[[Order]] is ~init~ and _E_ and _D_ have overlapping ranges.</li>
-        <li>For each pair (_E_, _D_) in EventSet(_execution_), _execution_.[[HappensBefore]] contains (_E_, _D_) if there is an event _F_ such that the pairs (_E_, _F_) and (_F_, _D_) are in _execution_.[[HappensBefore]].</li>
+        <li>For each pair (_E_, _D_) in is-agent-order-before in _execution_, _E_ happens-before _D_ in _execution_.</li>
+        <li>For each pair (_E_, _D_) in synchronizes-with in _execution_, _E_ happens-before _D_ in _execution_.</li>
+        <li>For each pair (_E_, _D_) in SharedDataBlockEventSet(_execution_), _E_ happens-before _D_ in _execution_ if _E_.[[Order]] is ~init~ and _E_ and _D_ have overlapping ranges.</li>
+        <li>For each pair (_E_, _D_) in EventSet(_execution_), _E_ happens-before _D_ in _execution_ if there is an event _F_ such that _E_ happens-before _F_ and _F_ happens-before _D_ in _execution_.</li>
```

Comment:
Here and in other places, there is type confusion for the `SharedDataBlockEventSet` and `EventSet` AOs. They return Sets of events, but they are being used as if they return Relations.

Reply:
I don't think so? I'm just enumerating all pairs of elements of a set. Relations are defined as pairs, but you can make pairs outside of relations.

Reply:
Would you be happier with `For each event _E_ in EventSet(_execution_) and _D_ in EventSet(_execution_)`?

Reply:
For that, I would say something like `For each event _E_ and _D_ such that EventSet(_execution_) contains both _E_ and _D_, ...`. But see my other suggestions for a more significant refactoring.